src/EmbedIO/Sessions/LocalSessionManager.cs

```diff
@@ -239,6 +239,18 @@ public void Delete(IHttpContext context, string id)
             context.Response.Cookies.Add(BuildSessionCookie(string.Empty));
         }
 
+        /// <inheritdoc />
+        public void Delete(IHttpContext context)
+        {
+            var id = GetSessionId(context);
+
+            lock (_sessions)
+            {
+                if (!string.IsNullOrEmpty(id) && _sessions.TryGetValue(id!, out _))
```

Comment:
The first check would be better moved before the lock, so if there is no session cookie we can bail out early and spare the lock altogether.